### PR TITLE
Release v2.5 of the Java Chaincode

### LINKED ISSUE #264: Fabric v2.5 Release
- [x] Update version numbers in build.gradle files to the required version
- [x] Update test, sample, and docs files to match the new version
- [x] Update the COMPATIBILITY.md file
- [x] Create a new release notes file
- [x] Update the CHANGELOG.md file

### PR DESCRIPTION
**Pull request by mbwhite**
Resolves #264 

Signed-off-by: Matthew B White <whitemat@uk.ibm.com>

### CODE REVIEW

File: COMPATIBILITY.md [hunk 1]
```diff
@@ -30,26 +32,30 @@ The key elements are : 
...
+Please check the [Dockerfile](./fabric-chaincode-docker/Dockerfile) that is used for the environment to see exactly how these versions are installed. 
 
 ### Supported Runtime communication with the Peer
  
-Subject to a suitable runtime environment, the 1.4.4 and 2.1.0 Java Libraries can used to communicate with a Fabric 2.1.0 or 1.4.4 Peer - with the level of functionality that is implied by the Fabric version in use. 
\ No newline at end of file
+Subject to a suitable runtime environment, the 1.4.4 and 2.5.0 Java Libraries can used to communicate with Fabric 2.5 and previous LTS versions (2.2.0/1.5.4) - with the level of functionality that is implied by the Fabric version in use and channel capabilities.
```
Comment (denyeart) [inline COMPATIBILITY.md:61]: I'm confused... 1.4.4 and 2.5.0 Fabric versions not Java Libraries right?
And what is 1.5.4?
And what do channel capabilities have to do with this?

File: release_notes/v2.5.0.txt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+Resolved Vulnerabilities
+------------------------
+none
+
+Known Issues & Workarounds
+--------------------------
+none
+
+Change Log
```
Comment (denyeart) [inline release_notes/v2.5.0.txt:24]: We can remove the changelog now.

File: COMPATIBILITY.md (hunk 1 above)
Comment (mbwhite) [reply COMPATIBILITY.md:61]: yes Fabric versions.... 1.5.4 was a typo.

Even if the peer is 2.5 etc.. if you've configured channel capability to v2, for example you can't use purgepd.

**Review by jt-nti — APPROVED** [review]
We'll need to update the doc site in a follow on PR

**Pull request merged**